README.md

```diff
@@ -163,6 +163,7 @@ with:
 | --- | --- |
 | `plan` | A human friendly output of the Terraform plan |
 | `plan_json` | A JSON representation of the Terraform plan |
+| `exit_code` | `"0"` succeeded with empty diff (no changes), `"1"` error, `"2"` succeeded with non-empty diff (changes present) |
```

Comment:
This is unduly coupled to the Terraform CLI and shell in general. Exit codes are an additional output for CLIs, where stdout/err are text and therefore typically not suitable for logic without other non-std programs (eg `jq`).

There's also a good reason why `-detailed-exitcode` is a flag an not the default. `exit 2` is still an error—the caller has to expect it (`set +e`) and then handle it (`if [ $? -eq 2 ]`). `tf.Plan` makes use of it:

https://github.com/hashicorp/terraform-exec/blob/v0.14.0/tfexec/plan.go#L94-L104

Exit codes don't really make sense as an interface in GitHub Actions, unless you are invoking a CLI that returns meaningful exit codes and will be passing along that info as an output for completeness.

GitHub already has some prior art here, with step conclusions/outcomes:

https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#steps-context

Since Actions doesn't face the same syntax/io limitations of shell, they use verbose status names rather than numeric codes.

If an action fails, it should exit 1, or anything non-zero for that matter. If the user wants to handle that failure, they set `continue-on-error: true` and then check whether `steps.foo.outcome == 'failure'`. That's already the case.

Checking for diffs is also already supported AFAICT. `steps.foo.plan == ''` will unambiguously check whether a plan was generated.